core/src/apps/common/sdcard.py

```diff
@@ -109,7 +109,7 @@ class SdCardUnavailable(wire.ProcessError):
                 fatfs.setlabel("TREZOR")
                 # mkfs and mount succeeded
                 return
-        except OSError:
+        except fatfs.FatFSError:
```

Comment:
I think we could allow retry here in case of OSError (i.e. the actual writing of bytes to the SD card fails).

Reply:
that does not raise OSError though; the call returns a C error result (see https://github.com/trezor/trezor-firmware/blob/master/core/embed/extmod/modtrezorio/modtrezorio-fatfs.h#L44-L85 ) which is then converted to ff.c error code, which converts to FatFSError.

Reply:
Oh yeah, thanks. I got confused by the errors we raise in modtrezorio-sdcard.h, not realizing the filesystem code accesses the SD card directly.

Reply:
Oh wait. But FilesystemWrapper::__enter__() calls sdcard.power_on(), which can raise OSError. I don't see that being caught anywhere ...

Reply:
resolved in dd189b6